packages/ui/src/components/Table/column-sort.composable.spec.ts

```diff
@@ -0,0 +1,133 @@
+  const column2 = shallowReactive({
+    prop: 'prop2',
+    sortable: true,
+    sortBy: ['prop1', 'prop2'],
+    sortMethod: null,
+    sortOrders: ['ascending', 'descending', null],
+  })
+
+  const column3 = shallowReactive({
+    prop: 'prop3',
+    sortable: true,
+    sortBy: '',
+    sortMethod: (a: typeof DATA[number], b: typeof DATA[number]) => a.prop3.length - b.prop3.length,
+    sortOrders: ['ascending', 'descending', null],
+  })
+
+  beforeEach(() => {
+    data.value = DATA
+  })
+
+  describe(`When composable initialized`, () => {
+    test(`Then 'sortedData' equal 'data'`, () => {
+      const { sortedData } = useColumnSort(data)
+      sortedData.value.forEach((row, i) => expect(toRaw(row)).toBe(DATA[i]))
+    })
+  })
+
+  describe(`When 'handleSortChange' called with first column`, () => {
+    test(`Then 'sortedData' becomes 'data' sorted with next sort order for this column`, () => {
+      const { sortState, sortedData, handleSortChange } = useColumnSort(data)
+      handleSortChange(column1 as ColumnApi)
+      expect(sortState.column).toBe(column1)
+      expect(sortState.order).toBe('ascending')
+      sortedData.value.forEach((row, i) => expect(toRaw(row)).toBe(DATA_PROP1_ASCENDING[i]))
+      handleSortChange(column1 as ColumnApi)
+      expect(sortState.column).toBe(column1)
+      expect(sortState.order).toBe('descending')
+      sortedData.value.forEach((row, i) => expect(toRaw(row)).toBe(DATA_PROP1_DESCENDING[i]))
+      handleSortChange(column1 as ColumnApi)
+      expect(sortState.column).toBe(column1)
+      expect(sortState.order).toBe(null)
+      sortedData.value.forEach((row, i) => expect(toRaw(row)).toBe(DATA[i]))
+    })
+  })
+
+  describe(`When 'handleSortChange' called with second column with array in 'sortBy' prop`, () => {
+    test(`Then 'sortedData' becomes 'data' sorted with all props in array`, () => {
+      const { sortState, sortedData, handleSortChange } = useColumnSort(data)
+      handleSortChange(column2 as ColumnApi)
+      expect(sortState.column).toBe(column2)
+      expect(sortState.order).toBe('ascending')
+      sortedData.value.forEach((row, i) => expect(toRaw(row)).toBe(DATA_PROP1_PROP2_ASCENDING[i]))
+    })
+  })
+
+  describe(`When 'handleSortChange' called with third column with special 'sortMethod'`, () => {
+    test(`Then 'sortedData' becomes 'data' sorted with this method`, () => {
+      const { sortState, sortedData, handleSortChange } = useColumnSort(data)
+      handleSortChange(column3 as ColumnApi)
+      expect(sortState.column).toBe(column3)
+      expect(sortState.order).toBe('ascending')
+      expect(toRaw(sortedData.value[0])).toBe(DATA_PROP3_ASCENDING[0])
+      expect(toRaw(sortedData.value[1])).toBe(DATA_PROP3_ASCENDING[1])
+      expect(toRaw(sortedData.value[2])).toBe(DATA_PROP3_ASCENDING[2])
+      expect(toRaw(sortedData.value[3])).toBe(DATA_PROP3_ASCENDING[3])
+      sortedData.value.forEach((row, i) => expect(toRaw(row)).toBe(DATA_PROP3_ASCENDING[i]))
```

Comment:
Could be much more readable and future support-friendly if it was tested via snapshot.